feat(dev): ignore self imports when computing update for invalidate

sapphi-red opened the pull request:
The self imports should be ignored when computing update for invalidate call because we already know that the module cannot self accept.

The react refresh wrapper plugin injects a self import.
https://github.com/rolldown/rolldown/blob/9883b9bf1e1b8c74e3c9f438af9fa7109e29b6ea/crates/rolldown_plugin_react_refresh_wrapper/src/lib.rs#L60

sapphi-red commented:
* **#6252** <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/6252?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/rolldown/rolldown/6252?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add the label <kbd>graphite: merge</kbd> to this PR to add it to the merge queue.

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/rolldown?ref=merge-queue-instructions-in-stack-comment&prId=6252).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

hyf0 reviewed (approved):
Without this, it results to full reload right?

hyf0 commented:
### Merge activity

* **Sep 17, 10:17 AM UTC**: The merge label 'graphite: merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=rolldown&repo=rolldown) once it meets the requirements.
* **Sep 17, 10:17 AM UTC**: `hyf0` added this pull request to the [Graphite merge queue](https://app.graphite.dev/merges?org=rolldown&repo=rolldown).
* **Sep 17, 10:26 AM UTC**: Merged by the [Graphite merge queue](https://app.graphite.dev/merges?org=rolldown&repo=rolldown).

sapphi-red committed:
feat(dev): ignore self imports when computing update for invalidate (#6252)

The self imports should be ignored when computing update for invalidate call because we already know that the module cannot self accept.

The react refresh wrapper plugin injects a self import.
https://github.com/rolldown/rolldown/blob/9883b9bf1e1b8c74e3c9f438af9fa7109e29b6ea/crates/rolldown_plugin_react_refresh_wrapper/src/lib.rs#L60

sapphi-red commented:
> Without this, it results to full reload right?

Yes, it results to full reload.
